homeassistant/components/melcloud/diagnostics.py

```diff
@@ -0,0 +1,27 @@
+"""Diagnostics support for MelCloud."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for the config entry."""
+    ent_reg = er.async_get(hass)
+    entities = [
+        entity.entity_id
+        for entity in er.async_entries_for_config_entry(ent_reg, entry.entry_id)
+    ]
+
+    entity_states = {entity: hass.states.get(entity) for entity in entities}
+
+    return {
+        "entry": entry.as_dict(),
```

Comment:
This entry contains a username and a token. These should not end up in diagnostics.

Diagnostic data must be stripped from personal data.

What would be interesting about the entry data in this case that it is useful to put it into the diagnostics?

../Frenck

Reply:
Oops, indeed! That has been properly redacted now.
Other than making it more _complete_ for diagnostics, there wasn't a particular property that I wanted to expose. Overall  I wanted to add diagnostics to help around after this [issue](https://github.com/home-assistant/core/issues/115987), where MELCloud silently failed. Having the opportunity to see the `last_reported` timestamp will help to see this behaviour, if help is being provided via Github.